### ISSUE TITLE
Bundle CSS from Antora default UI in plugin 

### ISSUE THREAD

**Issue by ahus1** [asciidoctor/asciidoctor-intellij-plugin#997]
## Why the new feature should be added

When editing content for Antora, users would like to have a preview similar to their Antora site.

There is documentation on how to reference [the CSS of the final Antora site](https://intellij-asciidoc-plugin.ahus1.de/docs/users-guide/features/advanced/stylesheets.html#replace-default-stylesheet), still it would be nicer to have something out-of-the box until the first site is published and as a fallback if the user is offline.

## How the new feature should work

When users edit content for an Antora module, they should see the preview with the Antora default UI theme and not the default Asciidoctor theme.

This would need to bundle both the CSS and relevant fonts. 

The default Antora UI theme doesn't support a dark mode AFAIK. Additional CSS needs to be added to make it dark-mode-ready.

This was originally suggested by @mojavelinux on the Antora community Zulip chat.

**Comment by ahus1** [comment]
Here a screenshot of a first local build, will work both in light and dark mode. This will work in both JCEF and JavaFX preview.
I'll update this issue once a pre-release is available.

![image](https://user-images.githubusercontent.com/3957921/156892446-1ea79798-aadd-43fb-8789-2c54a855577e.png)

![image](https://user-images.githubusercontent.com/3957921/156892404-92a846a8-c06a-4e92-80dc-eb90c447e9fb.png)

**Comment by ahus1** [comment]
Pre-release 0.37.7 is now available that includes these change.

To the ones that have a technical interest: This packages a copy of styles.css from the Antora default UI theme plus its fonts. Then it adds a minimal set of CSS classes for the light preview, and some more CSS classes for the dark theme. Compared to the default Antora the plugin adds also font awesome, so that `icon:XX[]` works like in Asciidoctor. 

See here for the files: https://github.com/asciidoctor/asciidoctor-intellij-plugin/tree/main/src/main/resources/org/asciidoc/intellij/editor/javafx/antora

The dark theme might need some additional tuning, please comment or create PRs. 

The pre-release of the plugin is available from [GitHub releases](https://intellij-asciidoc-plugin.ahus1.de/docs/contributors-guide/user/installing-a-preview-release.html) and the [IntelliJ AsciiDoc EAP repository](https://intellij-asciidoc-plugin.ahus1.de/docs/contributors-guide/user/add-eap-repository-to-ide.html).

**Comment by RayOffiah** [comment]
Works like a charm!

I don't know how helpful this is, as it's just a bit of a hack really. 

I wanted the preview to look like our Antora site, so I created an .asciidoctorconfig file

 ```
:stylesdir: {asciidoctorconfigdir}/../local-css
:stylesheet: local.css
```

And then in my local.css file, I point to the site.css file on the Antora site.

```
@import url('https://docs.mysite.com/_/css/site.css'); // a local build should work too.
```

I was thinking that if someone build a local preview Antora site, you could have a project-specific setting to point to the preview src and use that for styling the editor.

**Comment by ahus1** [comment]
Hi @RayOffiah - thanks for giving it a try. This gives me the confidence to promote it to a regular release so everyone else will download it. 

The "hack" you describe is the current solution on how to use an external stylesheet - even after this update. 

The `local.css` adds the possibility to override some of the CSS classes of the final site. A slightly shorter but less flexible approach is to have only the `.asciidoctorconfig` file with the following content: 

```
:linkcss:
:stylesdir: https://docs.mysite.com/_/css/
:stylesheet: site.css
```

In this setup the attribute `linkcss` allows users to specify the URL directly. It also activates another feature of the plugin where will fallback to the default CSS when the remote CSS is not available (for example when working in offline mode or when the server is down). 

There are some [docs around this](https://intellij-asciidoc-plugin.ahus1.de/docs/users-guide/features/advanced/stylesheets.html#remote-stylesheet). Thinking about it, I should add this to the Antora documentation as well with the Antora default settings. 

Maybe a future version will automatically discover a stylesheet from a local playbook. Let's see.

**Comment by RayOffiah** [comment]
> The "hack" you describe is the current solution on how to use an external stylesheet - even after this update.

Sorry, I phrased that badly.

The 'hack' wasn't the .asciidoctorconfig part; it was using the site.css file from a Antora site, which is probably less than ideal, but works pretty well.

**Comment by ahus1** [comment]
Hi @RayOffiah - "hack" has quite a positive meaning for me, as of [Merriam-Webster](https://www.merriam-webster.com/dictionary/hack): "a usually creatively improvised solution to a computer hardware or programming problem or limitation", so no offense meant or taken here.

If it does what you intended in a reliable way, there is usually no need to change it. If more users need it, then it's getting polished and becomes a "feature", not sure how and when this will happen for this one. 

So 👏 👏 to your "creatively improvised solution".

**Comment by ahus1** [comment]
This has been available from pre-release 0.37.7 without any obvious need for re-work. Closing this issue.
